js/modules/cohortbuilder/CohortExpression.js

```diff
@@ -27,7 +28,10 @@ define(function (require, exports) {
 			return CriteriaTypes.GetCriteriaFromObject(criteria, self.ConceptSets);
 		}));
 		self.CollapseSettings = {CollapseType: ko.observable(data.CollapseSettings && data.CollapseSettings.CollapseType || "ERA"), EraPad: ko.observable(data.CollapseSettings && data.CollapseSettings.EraPad || 0 ) }
-		
+		self.CensorWindow = ko.observable(new Period({
+			StartDate: data.CensorWindow ? data.CensorWindow.StartDate : null,
+			EndDate: data.CensorWindow ? data.CensorWindow.EndDate : null,
+		}));
```

Comment:
You don't need to construct the JSON in the new Period() call. You can pass the element directly from the raw data:

```
  self.CensorWindow = ko.observable(data.CensorWindow && new Period(data.CensorWindow));
```
This will only allocate a Period object if CensorWindow is not null.